src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java

```diff
@@ -242,6 +245,15 @@ private int memory(AppDeploymentRequest request) {
 			request.getDeploymentProperties().getOrDefault(MEMORY_PROPERTY_KEY, valueOf(deploymentProperties.getMemory())));
 	}
 
+	private ApplicationHealthCheck healthCheck(AppDeploymentRequest request) {
+		ApplicationHealthCheck result = deploymentProperties.getHealthCheck();
+		String override = request.getDeploymentProperties().get(HEALTHCHECK_PROPERTY_KEY);
+		if (override != null) {
+			result = ApplicationHealthCheck.valueOf(override.toUpperCase());
```

Comment:
The doc will go in the server project. Waiting for https://github.com/spring-cloud/spring-cloud-dataflow-server-cloudfoundry/pull/172 to be merged (feel free to go ahead and merge it if you can), as it's likely to have conflicts.

Will provide a better error message here